### ISSUE TITLE
Eth-Precision coupon value error

### ISSUE THREAD

**Issue by jjeffryes** [OpenBazaar/openbazaar-desktop#1824]
When loading a listing, I see the following error in the console:

`Unable to convert coupon.bigPriceDiscount:  from an integer to a decimal: The value must be provided as one of: number, non NaN bigNumber.js instance, string based representation of a number`

It doesn't seem to block any functionality, I'm still able to view the listing and make a purchase.

**Comment by rmisio** [comment]
It happens for every listing? If not, can you please post the listing data?

**Comment by jjeffryes** [comment]
It only happens on listings that have coupons.

ex: ob://QmcF2eG2tpRCZddxi8oCnpk5E8pHyRGp3nFbFzPMBGpyGt/store/test-phys-eth-precision

**Comment by jjeffryes** [comment]
Here's the listing data:

```
{
    "listing": {
        "slug": "test-phys-eth-precision",
        "vendorID": {
            "peerID": "QmcF2eG2tpRCZddxi8oCnpk5E8pHyRGp3nFbFzPMBGpyGt",
            "handle": "",
            "pubkeys": {
                "identity": "CAESIEUwMFbeuzGvRyD3H56TRplxUVKXrBheuoAwIhmZY7pA",
                "bitcoin": "A1jFMiVs4OIWXWs7ExQTKmPYiVpgqjwE+YDkTf+0UFWf"
            },
            "bitcoinSig": "MEQCIEiOqmD/0ZSdWy7GuSqxHDYxNeRWxIqWsuShaczYkZEHAiAQi8w51qeoCkbMAMzjE1KU4U2vEMcHRM57kWhlODLVVw=="
        },
        "metadata": {
            "version": 5,
            "contractType": "PHYSICAL_GOOD",
            "format": "FIXED_PRICE",
            "expiry": "2037-12-31T05:00:00.000Z",
            "acceptedCurrencies": [
                "ETH",
                "BTC",
                "BCH",
                "ZEC",
                "LTC"
            ],
            "pricingCurrency": "",
            "language": "",
            "escrowTimeoutHours": 1080,
            "coinType": "",
            "coinDivisibility": 0,
            "priceModifier": 0
        },
        "item": {
            "title": "test phys eth precision",
            "description": "Just a test for the ETH precision update",
            "processingTime": "",
            "price": 0,
            "nsfw": false,
            "tags": [
                "test"
            ],
            "images": [
                {
                    "filename": "The_Four_Seasons___Autumn_by_Zhaana.jpg",
                    "original": "QmUs2PLWQWzRZZmizmY5RBvjSEummyuX28AXCceKnC9vrG",
                    "large": "QmT8Hvbe5ZP2JMc4uxdQDu3WuPyFF3HFXt1SjWu4Ng5L1b",
                    "medium": "QmUMhUZBB5QQQBrS6phnMyGKsYYiQi1CCQ3B3m3LwrGLie",
                    "small": "QmWCbJvPmqf6ZuCAv7n7mbeAf6z3yMYj1ts4LY4Rky4meQ",
                    "tiny": "Qmc9VJQ73DGapYXbnYLRohKW8WhvgE74ekL6Q4DYuMHXru"
                }
            ],
            "categories": [
                "test"
            ],
            "grams": 0,
            "condition": "USED_POOR",
            "options": [
                {
                    "name": "color",
                    "description": "",
                    "variants": [
                        {
                            "name": "red"
                        },
                        {
                            "name": "green"
                        },
                        {
                            "name": "blue"
                        }
                    ]
                },
                {
                    "name": "size",
                    "description": "",
                    "variants": [
                        {
                            "name": "small"
                        },
                        {
                            "name": "large"
                        }
                    ]
                }
            ],
            "skus": [
                {
                    "variantCombo": [
                        0,
                        0
                    ],
                    "productID": "1",
                    "surcharge": 0,
                    "quantity": 0,
                    "bigSurcharge": "0",
                    "bigQuantity": "1"
                },
                {
                    "variantCombo": [
                        0,
                        1
                    ],
                    "productID": "2",
                    "surcharge": 0,
                    "quantity": 0,
                    "bigSurcharge": "1",
                    "bigQuantity": "1"
                },
                {
                    "variantCombo": [
                        1,
                        0
                    ],
                    "productID": "3",
                    "surcharge": 0,
                    "quantity": 0,
                    "bigSurcharge": "0",
                    "bigQuantity": "1"
                },
                {
                    "variantCombo": [
                        1,
                        1
                    ],
                    "productID": "4",
                    "surcharge": 0,
                    "quantity": 0,
                    "bigSurcharge": "10",
                    "bigQuantity": "1"
                },
                {
                    "variantCombo": [
                        2,
                        0
                    ],
                    "productID": "5",
                    "surcharge": 0,
                    "quantity": 0,
                    "bigSurcharge": "0",
                    "bigQuantity": "-1"
                },
                {
                    "variantCombo": [
                        2,
                        1
                    ],
                    "productID": "6",
                    "surcharge": 0,
                    "quantity": 0,
                    "bigSurcharge": "1",
                    "bigQuantity": "-1"
                }
            ],
            "priceModifier": 0,
            "bigPrice": "1",
            "priceCurrency": {
                "code": "ZWL",
                "divisibility": 2
            }
        },
        "shippingOptions": [
            {
                "name": "test option title",
                "type": "FIXED_PRICE",
                "regions": [
                    "ALAND_ISLANDS"
                ],
                "services": [
                    {
                        "name": "test service",
                        "price": 0,
                        "estimatedDelivery": "test delivery time",
                        "additionalItemPrice": 0,
                        "bigPrice": "0",
                        "bigAdditionalItemPrice": "1"
                    }
                ]
            }
        ],
        "coupons": [
            {
                "title": "test",
                "discountCode": "test",
                "percentDiscount": 10,
                "priceDiscount": 0,
                "bigPriceDiscount": ""
            }
        ],
        "moderators": [
            "QmPGAF6PvqcgjF1Tcc3HbQLvieTTMaxqJfySet4jB6jLq3"
        ],
        "termsAndConditions": "test terms",
        "refundPolicy": "test return policy"
    },
    "hash": "QmRvnWubsa3M25gchJfyviPRbhK9q9LbASRF5raKR88Wuz",
    "signature": "c8uov0LDDQ+ffLOFLl5RA0U3vchcazcWDi8xtosxFTbDt8dL3LG35JVXoJB5bZNHuLJHPYxTJnNEJHNz6tSPCQ=="
}
```

**Comment by rmisio** [comment]
fixed via 1cb3cbacb4aa12da1011220507b8886f4bb469c8
